js/src/components/horizontal-stacked-bar/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Legend from './legend';
+import Bars from './bars';
+import './index.scss';
+
+/**
+ * @typedef {Object} Segment
+ * @property {string|number} id - Unique identifier for the segment
+ * @property {string} label - Display label for the segment
+ * @property {string|number} value - Numeric value determining segment size
```

Comment:
I see that you've defined value to be either a string or a number. Seems like that could cause errors when calculating percentages and totals in the `HorizontalStackedBar`. Can we add some validation logic and ensure tests include cases for when invalid values are passed as segments?

Reply:
Sure! I've filtered out any non numeric values from the segments. Any segments with a non numeric value will be omitted from the chart.